docs/transport.rst

```diff
@@ -0,0 +1,214 @@
+
+UDP Server Example
+^^^^^^^^^^^^^^^^^^
+
+As an example, this subsection describes how the UDP server is implemented in *eProsima Micro XRCE-DDS Agent*.
+The figure below shows the *Agent* transport architecture for the UDP servers.
+
+.. image:: images/agent_transport_architecture.svg
+
+At the top of this architecture, there is a ``Server`` interface (Server Layer).
+This ``Server`` interface has the following pure virtual functions:
+
+.. code-block:: cpp
+
+    /* Transport Layer */
+    virtual void on_create_client(EndPoint* source, const dds::xrce::ClientKey& client_key) = 0;
+    virtual void on_delete_client(EndPoint* source) = 0;
+    virtual const dds::xrce::ClientKey get_client_key(EndPoint* source) = 0;
+    virtual std::unique_ptr<EndPoint> get_source(const dds::xrce::ClientKey& client_key) = 0;
+
+    /* Platform Layer */
+    virtual bool init() = 0;
+    virtual bool close() = 0;
+    virtual bool recv_message(InputPacket& input_packet, int timeout) = 0;
+    virtual bool send_message(OutputPacket output_packet) = 0;
+    virtual int get_error() = 0;
+
+The first 4 virtual function are transport specific (Transport Layer).
+These functions are override by the ``UDPServerBase`` class which is in charge of managing the sessions between *Clients* and the *Agent*.
+
+On the other hand, the last 5 virtual function are platform specific (Platform Layer).
+These functions are override by the ``UDPServerLinux`` and ``UDPServerWindows`` for Linux and Windows systems respectively.
+
+Client Transport Architecture
+-----------------------------
+
+The *Client* transport architecture is analogous to the *Agent* architecture.
+There are also 3 different layers but instead of the Server Layer, there is a Session Layer.
+
+.. image:: images/client_transport_stack.svg
+
+* **Session Layer**: implements the XRCE protocol logic and it only knowns about sending and receiving messages.
+* **Transport Layer**: implements the sending and receiving **message functions** for each transport protocol, calling to the Platform Layer functions.
+  This layer only knows about sending and receiving messages through a given transport protocol.
+* **Platform Layer**: implements the sending and receiving **data functions** for each platform.
+  This layer only knows about sending and receiving raw data through a given transport in a given platform.
+
+UDP Transport Example
+^^^^^^^^^^^^^^^^^^^^^
+
+As an example, this subsection describes how the UDP transport is implemented in *eProsima Micro XRCE-DDS Client*.
+The figure below shows the *Client* transport architecture for the UDP transport.
+
+.. image:: images/client_transport_architecture.svg
+
+Similar to the *Agent* architecture, there is also an interface, ``uxrCommunication``, whose functions are call from the Session Layer.
```

Comment:
Because is a *c struct*, maybe talk about its attributes instead of *functions*?
Besides the *functions*, you always have the *mtu* attribute, that is not a function